modules/core/src/lifecycle/prop-types.ts

```diff
@@ -1,77 +1,149 @@
 import {createTexture, destroyTexture} from '../utils/texture';
 
+import type Component from './component';
+
+export interface PropType {
```

Comment:
Why is this `interface` but all the others `type`? Any reason not to use `type` here as well?